core/src/main/java/com/linecorp/armeria/server/DefaultServerConfig.java

```diff
@@ -312,23 +310,13 @@ private static Mapping<String, VirtualHost> buildDomainMapping(VirtualHost defau
         return mappingBuilder.build();
     }
 
-    private static ScheduledExecutorService monitorBlockingTaskExecutor(ScheduledExecutorService executor,
-                                                                        MeterRegistry meterRegistry) {
-        final ScheduledExecutorService unwrappedExecutor;
-        if (executor instanceof BlockingTaskExecutor) {
-            unwrappedExecutor = ((BlockingTaskExecutor) executor).unwrap();
-        } else {
-            unwrappedExecutor = executor;
-        }
-
+    private static BlockingTaskExecutor monitorBlockingTaskExecutor(BlockingTaskExecutor executor,
+                                                                    MeterRegistry meterRegistry) {
         new ExecutorServiceMetrics(
-                unwrappedExecutor,
+                executor.unwrap(),
                 "blockingTaskExecutor", "armeria", ImmutableList.of())
                 .bindTo(meterRegistry);
-        executor = new TimedScheduledExecutorService(meterRegistry, executor,
-                                                     "blockingTaskExecutor", "armeria.",
-                                                     ImmutableList.of());
```

Comment:
Some users might be surprised that `TimedScheduledExecutorService`-related metrics will disappear from their monitoring dashboard. What do you think of adding `TimedBlockingTaskExecutor` by extending `TimedScheduledExecutorService`?

Reply:
It's very likely that those metrics are all zero already because the executor returned by `ctx.blockingTaskExecutor()` is not a `TimedScheduledExecutorService`. 😱 How about having a separate PR for 1.24?